lib/rules/unsafe-to-chain-command.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Actions should be in the end of chains, not in the middle',
+      category: 'Possible Errors',
+      recommended: true,
+      url: 'https://docs.cypress.io/guides/core-concepts/retry-ability#Actions-should-be-at-the-end-of-chains-not-the-middle',
+    },
+    schema: [],
+    messages: {
+      unexpected: 'It is unsafe to chain further commands that rely on the subject after this command. It is best to split the chain, chaining again from `cy.` in a next command line.',
+    },
+  },
+  create (context) {
+    return {
+      CallExpression (node) {
+        if (isRootCypress(node) && isActionUnsafeToChain(node) && node.parent.type === 'MemberExpression') {
+          context.report({ node, messageId: 'unexpected' })
+        }
+      },
+    }
+  },
+}
+
+function isRootCypress (node) {
+  while (node.type === 'CallExpression') {
+    if (node.callee.type !== 'MemberExpression') return false
+
+    if (node.callee.object.type === 'Identifier' &&
+        node.callee.object.name === 'cy') {
+      return true
+    }
+
+    node = node.callee.object
+  }
+
+  return false
+}
+
+function isActionUnsafeToChain (node) {
+  // commands listed in the documentation with text: 'It is unsafe to chain further commands that rely on the subject after xxx'
+  const unsafeToChainActions = ['blur', 'clear', 'click', 'check', 'dblclick', 'each', 'focus', 'rightclick', 'screenshot', 'scrollIntoView', 'scrollTo', 'select', 'selectFile', 'spread', 'submit', 'type', 'trigger', 'uncheck', 'wait', 'within']
```

Comment:
Minor note: In Cypress 12.0.0 - 12.2.0, `should` and `and` are also unsafe. This is a bug we'll be fixing ASAP, likely in 12.3.0 - see https://github.com/cypress-io/cypress/issues/25134 for discussion and details.

Reply:
Or maybe wait command should be considered safe to chain (documentation was updated after this PR?). If that's the way to go, we just need to merge this PR to get it removed: https://github.com/cypress-io/eslint-plugin-cypress/pull/128

Reply:
Yes. very confusing I am using this a lot

      cy.getBySel('post.title').find('input').clear().type(newPost.title)
      but not my ESLINT started to show warning